debug_script.py

```diff
@@ -0,0 +1,52 @@
+import numpy as np
+from numpy.linalg import norm
+
+from skglm.penalties.block_separable import WeightedGroupL1
+from skglm.datafits.group import QuadraticGroup
+from skglm.solvers.group_bcd_solver import bcd_solver
+
+from skglm.utils import grp_converter, make_correlated_data
+
+
+def _generate_random_grp(n_groups, n_features, random_state=123654):
```

Comment:
no need to exposte the parameter random_state. One parameter fewer makes the code easier, you can hardcode it